File: Julia/Problem004.jl
function isPalindrome(n)
	return cmp(string(n), reverse(string(n))) == 0
end

# Parameters:
D = 3
###############

for i ∈ (10^D - 1)^2 : -1 : (10^(2*D - 2))
	if isPalindrome(i)
		for j ∈ (10^D - 1) : -1 : (10^(D - 1))
			if i % j == 0
				if (10^(D - 1)) <= i / j && i / j <= (10^D - 1)
					println(i)
					return
				end
			end
		end
	end
end

File: Python/Problem004.py
def isPalindrome(num):
    reverse = int(str(num)[::-1])
    return reverse == num


d = 3

for i in range((10**d - 1)**2, 0, -1):
    if isPalindrome(i):
        for j in range((10**d - 1), (10**(d - 1) - 1), -1):
            if i % j == 0:
                if (10**(d - 1) - 1) < i / j <= (10**d - 1):
                    print(i)
                    exit()
